tests/integration-tests/tests/create/test_create.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+import logging
+
+import pytest
+from remote_command_executor import RemoteCommandExecutor
+
+from tests.common.assertions import assert_errors_in_logs
+from tests.common.utils import retrieve_latest_ami
+
+
+@pytest.mark.dimensions("eu-central-1", "c5.xlarge", "ubuntu1804", "slurm")
+def test_create_wrong_os(region, os, pcluster_config_reader, clusters_factory, test_datadir, scheduler, instance):
+    """Test error message when os provide is different from the os of custom AMI"""
+    # ubuntu1804 is specified in the config file but an AMI of centos7 is provided
+    wrong_os = "centos7"
+    custom_ami = retrieve_latest_ami(region, wrong_os)
```

Comment:
should we use pcluster ami rather than vanilla ami to test pcluster create to ensure common cases?

Reply:
Common case should be an unofficial AMI. I might be wrong, official AMIs are usually passed correctly since it is the case when custom_ami is not specified in the config file

Reply:
Just a note, I don't think run time bake is fully supported today. In other words, if we start with a vanilla AMI there is chance that cluster might fail because of some other issue and not OS validation.
Of course OS validation should be the first thing that errors, so we don't necessarily need to change this, but just something to keep in mind

I would consider using a strong assertion to make sure cluster fails because of OS validation, please think about if asserting on `fr"custom AMI.+{wrong_os}.+base.+os.+{os}"` is enough. If it is, I am ok with keeping things as is

Reply:
what about we choose remarkable ami like deep learning ami for os validation test. Even if we use unofficial ami here, I think is remarkable ami  better than vanilla to ensure common cases and avoid other failure like volume is not enough?

Reply:
Done